src/pages/Search/SearchFiltersCardPage.tsx

```diff
@@ -0,0 +1,124 @@
+import React, {useCallback, useMemo, useState} from 'react';
+import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
+import Button from '@components/Button';
+import type {FormOnyxValues} from '@components/Form/types';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import getBankIcon from '@components/Icon/BankIcons';
+import type {BankName} from '@components/Icon/BankIconsUtils';
+import ScreenWrapper from '@components/ScreenWrapper';
+import SelectionList from '@components/SelectionList';
+import CardListItem from '@components/SelectionList/CardListItem';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import type {CategorySection} from '@libs/OptionsListUtils';
+import type {OptionData} from '@libs/ReportUtils';
+import Navigation from '@navigation/Navigation';
+import * as SearchActions from '@userActions/Search';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+
+function SearchFiltersCardPage() {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const [cardList] = useOnyx(ONYXKEYS.CARD_LIST);
+    const [searchAdvancedFiltersForm] = useOnyx(ONYXKEYS.FORMS.SEARCH_ADVANCED_FILTERS_FORM);
+    const currentCards = searchAdvancedFiltersForm?.cardID;
+    const [newCards, setNewCards] = useState(currentCards ?? []);
+
+    const sections = useMemo(() => {
+        const newSections: CategorySection[] = [];
+        const cards = Object.values(cardList ?? {})
+            .filter((card) => card.cardID)
+            .sort((a, b) => a.bank.localeCompare(b.bank))
+            .map((card) => {
+                const icon = getBankIcon({bankName: card.bank as BankName, isCard: true, styles});
+
+                return {
+                    text: card.bank,
+                    keyForList: card.cardID.toString(),
+                    isSelected: newCards.includes(card.cardID.toString()),
+                    bankIcon: icon,
+                };
+            });
+        newSections.push({
+            title: undefined,
+            data: cards,
+            shouldShow: cards.length > 0,
+        });
+        return newSections;
+    }, [cardList, styles, newCards]);
+
+    const updateCard = useCallback((values: Partial<FormOnyxValues<typeof ONYXKEYS.FORMS.SEARCH_ADVANCED_FILTERS_FORM>>) => {
+        SearchActions.updateAdvancedFilters(values);
+    }, []);
+
+    const handleConfirmSelection = useCallback(() => {
+        updateCard({
+            cardID: newCards,
+        });
```

Comment:
your `updateCard` function is just a oneliner, so it's really not needed in here.
Just call `updateAdvancedFilters()` directly in this line